tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -399,3 +476,67 @@ func testAzureSSHConnection(t *testing.T) {
 	err = session.Run("dir")
 	require.NoErrorf(t, err, "failed to communicate vis SSH")
 }
+
+// test if the windows VM is behind the worker node load balancer
+func testAzureVMLoadbalancer(t *testing.T) {
+	ctx := context.Background()
+
+	// create a map to keep track of all the ip configurations associated with the backend pool .
+	ipCSet := make(map[string]struct{})
+	// filler value for map
+	var exists = struct{}{}
+
+	// We assume that BackendPoolID for worker vm follows the following format
+	// /subscriptions/$SUBSCRIPTID/resourceGroups/$INFRAID-rg/providers/Microsoft.Network/loadBalancers/$INFRAID/backendAddressPools/$INFRAID
+	// We assume that loadbalancer and backendAddressPools both have $INFRAID as name
+	bapInfo, err := azureInfo.lbBapClient.Get(ctx, azureInfo.resourceGroupName, azureInfo.infraID, azureInfo.infraID)
+	if err != nil {
+		t.Fatalf("failed to get backendAddressPool information: %v", err)
+	}
+	if bapInfo.BackendIPConfigurations == nil {
+		t.Fatalf("failed to get backend address pool configuration")
+	}
+	backendIPC := *bapInfo.BackendIPConfigurations
+	if len(backendIPC) < 1 {
+		t.Fatalf("empty backend address pool configuration for %s", *bapInfo.ID)
+	}
+	// get all the ip configurations associated with the backend address pool
+	for _, ipC := range backendIPC {
+		ipCSet[extractResourceName(*ipC.ID)] = exists
+	}
+
+	// check if all the vm are present in the backend address pool associated with the worker node.
+	for _, vmName := range instanceIDs {
```

Comment:
Why do we need to loop over all the worker instances? The only instance we created is the Windows VM and we need to check if our Windows node is behind the LB. This way we can be more specific and targeted in our test.

Reply:
i thought we decided on testing all instances in design meet. i can change this to test single vm if needed to. 
Also, what happens in the scenario where the user created 2 VMs and started the test after that?

Reply:
@ravisantoshgudimetla `instanceIDs` here is a global variable holding only the Windows VMs created as part of the e2e test and not all the worker nodes.

Reply:
> 

@ravisantoshgudimetla instanceIDs here is a global variable holding only the Windows VMs created as part of the e2e test and not all the worker nodes.



This is where I got confused, thanks for the  clarification.